internal/observability/graph.go

```diff
@@ -105,6 +105,7 @@ func (g *graph) begin(ctx context.Context, rpcType transport.Type, direction dir
 	d.Add(req.Procedure)
 	d.Add(req.RoutingKey)
 	d.Add(req.RoutingDelegate)
+	d.Add(req.CallerProcedure)
```

Comment:
I'm concerned about creating a new edge for every different callProcedure and other keys combinations (routing delegate, routing key) as this will increase edge cache size significantly for services with different routing-key/routing-delegate and also it might put pressure on the m3 collector as it's going to receive too unaggregated metrics. 

An alternative option might be to pass source-procedure in the zap extra fields accepted by the `endWithAppError` method, what do you think?

https://github.com/yarpc/yarpc-go/blob/9d0cab62f323aac0e60616fd3952d5b36ca85479/internal/observability/call.go#L264

Reply:
Very good point! given that this change is just for logging, we don't want to include CallerProcedure in to the key generation.